galsim/phase_psf.py

```diff
@@ -0,0 +1,1399 @@
+    argument.  Each screen in the list computes its own preferred sampling rate and the
+    PhaseScreenList appropriately aggregates these. This last option also requires that a wavelength
+    `lam` be specified, and is particularly helpful for creating PSFs derived from turbulent
+    atmospheric screens.
+
+    Finally, when specifying the pupil geometrically, Aperture may choose to make a small adjustment
+    to `pupil_plane_scale` in order to produce an array with a good size for FFTs.  If your
+    application depends on knowing the size and scale used with the Fourier optics framework, you
+    can obtain these from the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale`
+    attributes.
+
+    Pupil image specification
+    --------------------------
+
+    The second way to specify the pupil plane configuration is by passing in an image of it.  This
+    can be useful, for example, if the struts are not evenly spaced or are not radially directed, as
+    is assumed by the simple model for struts described above.  In this case, an exception is raised
+    if keywords related to struts are also given.  On the other hand, the `obscuration` keyword is
+    still used to ensure that the PSF images are not aliased, though it is ignored during the actual
+    construction of the pupil plane illumination pattern.  The `pupil_plane_im` that is passed in
+    can be rotated during internal calculations by specifying a `pupil_angle` keyword.  Also note
+    that given how the `obscuration` keyword is used, the obscuration in the supplied pupil plane
+    image should indeed be reasonably circular rather than having an arbitrary shape; however, minor
+    deviations from a perfect circle do not cause significant problems.
+
+    If you choose to pass in a pupil plane image, it must be a square array in which the image of
+    the pupil is centered.  The areas that are illuminated should have some value >0, and the other
+    areas should have a value of precisely zero.  Based on what the Aperture class determines is a
+    good PSF sampling interval, the image of the pupil plane that is passed in might be zero-padded
+    during internal calculations.  (The pupil plane array size and scale values can be accessed via
+    the `Aperture.pupil_plane_size` and `Aperture.pupil_plane_scale` attributes.) The pixel scale of
+    the pupil plane can be specified in one of three ways.  In descending order of priority, these
+    are:
+      1.  The `pupil_plane_scale` keyword argument (units are meters).
+      2.  The `pupil_plane_im.scale` attribute (units are meters).
+      3.  If (1) and (2) are both None, then the scale will be inferred by assuming that the
+          illuminated pixel farthest from the image center is at a physical distance of self.diam/2.
+
+    The `pupil_plane_size` and `lam` keywords are both ignored when constructing an Aperture from an
+    image.
+
+    @param diam             Aperture diameter in meters.
```

Comment:
I took another look at `Aperture` and `PhaseScreenPSF` to see if adding a `lam_over_diam` kwarg would make sense.  I think the answer is probably no.  `lam_over_diam` works for optics, but for the atmosphere, `lam_over_r0` is actually more relevant (though under this framework, and in contrast to `galsim.Kolmogorov`, you still need to know the Aperture diameter).  So I think keeping things separate is actually the clearest way to proceed here.